File: app/src/main/java/com/lumiyaviewer/lumiya/ui/render/vr/VrPose.java
package com.lumiyaviewer.lumiya.ui.render.vr;

public interface VrPose {
    void getQuaternion(float[] out, int offset);

    void getTranslation(float[] out, int offset);

    void getHeadView(float[] out, int offset);

    void getEulerAngles(float[] out, int offset);

    void getRightVector(float[] out, int offset);
}

File: app/src/main/java/com/lumiyaviewer/lumiya/ui/render/vr/VrPose.kt
package com.lumiyaviewer.lumiya.ui.render.vr

interface VrPose {
    fun getQuaternion(out: FloatArray, offset: Int)
    fun getTranslation(out: FloatArray, offset: Int)
    fun getHeadView(out: FloatArray, offset: Int)
    fun getEulerAngles(out: FloatArray, offset: Int)
    fun getRightVector(out: FloatArray, offset: Int)
}
